### DISCUSSION TITLE
How to sent Content-Length on a HEAD response

### DISCUSSION THREAD

**Discussion by ajmichels** [koajs/koa#1827] [Q&A] [answered]
I am trying to sent the Content-Length header of a HEAD method response but it seems like Koa it automatically removing the header when the body is set to `null`. I am sure there is some trick to this that I am just not familiar with but I can't figure out what it is.

```javascript
export async function head(ctx, next) {
    ctx.set('Content-Length', Buffer.from(JSON.stringify(ctx.body)).length);
    ctx.body = null;
    await next();
}
```

*Note: `ctx.body` is originally being set in a middleware function that runs before this one.*

**Comment by jonathanong** [comment]
try setting the content length after setting the body

**Comment by jonathanong** [comment, marked as answer]
additionally, since you already set the body, `koa` will know not to send it on HEAD requests, so you actually don't need to do anything
